src/components/Admin/ManageQuestions.js

```diff
@@ -0,0 +1,133 @@
+import React, { Component, Fragment } from "react";
+import styled from "styled-components";
+
+import Button from "react-bootstrap/Button";
+import Col from "react-bootstrap/Col";
+import Row from "react-bootstrap/Row";
+import Toast from "react-bootstrap/Toast";
+import AdminLayout from "./AdminLayout";
+import QuestionDisplay from "./QuestionDisplay";
+import AddQuestion from "./AddQuestion";
+
+import { AuthUserContext, withFirebase } from "upe-react-components";
+
+import Loader from "../Loader";
+import { Container } from "../../styles/global";
+
+const StyledDiv = styled.div`
+  text-align: right;
+`;
+
+class ManagerQuestions extends Component {
+  constructor(props) {
+    super(props);
+
+    this.updatePage = this.updatePage.bind(this);
+    this.toggleAdd = this.toggleAdd.bind(this);
+  }
+
+  _initFirebase = false;
+  state = {
+    questionList: null,
+    settings: null,
+    loading: true,
+    error: null,
+    addQuestion: false,
+  };
+  static contextType = AuthUserContext;
+  unsub = null;
+
+  componentDidMount() {
+    if (this.props.firebase && !this._initFirebase) this.loadSettings();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.firebase && !this._initFirebase) this.loadSettings();
+  }
+
+  componentWillUnmount() {
+    if (typeof this.unsub === "function") this.unsub();
+  }
+
+  updatePage = () => {
+    this.setState({ addQuestion: false });
+    this.loadSettings();
+  };
+
+  loadSettings = async () => {
+    this._initFirebase = true;
+
+    this.props.firebase
+      .questions()
+      .get()
+      .then((querySnapshot) => {
+        const questionList = querySnapshot.docs.map((doc) => {
+          return { id: doc.id, ...doc.data() };
+        });
+        this.setState({ questionList, loading: false }, () => {
+          console.log("Questions Loaded");
+        });
+      });
+  };
+
+  toggleAdd = () => {
+    this.setState({ addQuestion: !this.state.addQuestion });
+  };
+
+  render() {
+    const { loading, error, questionList, addQuestion } = this.state;
+
+    if (loading) return <Loader />;
+    if (error)
+      return (
+        <Container flexdirection="column">
+          <h1>{error}</h1>
+        </Container>
+      );
+
+    const authUser = this.context;
+
+    const Questions = () => {
+      return (
+        <Row>
+          {Object.entries(questionList).map((question) => (
```

Comment:
`questionList` is already an array, we shouldn't be doing Object.entries here. Also, we're saving the question ID from firebase, so we should just use the question id as the key and pass the entire question as props through spread. i.e.
```
{questionList.map((question) => (
  <QuestionDisplay
    updateFunc={this.updatePage}
    key={question.id}
    {...question}
  />
))}
```

This will require changing QuestionDisplay to work with it. It should become a function component, as it's complexity doesn't require a class, and making it a function component will make it simpler to work with passing props (no need for this.props).

Reply:
Resolved in commit 328335366bf582d828349cde311f9f15794bbca6